cmd/channel/distributed/dispatcher/main.go

```diff
@@ -224,3 +223,27 @@ func configMapObserver(ctx context.Context, configMap *corev1.ConfigMap) {
 		dispatcher = newDispatcher
 	}
 }
+
+// secretObserver is the callback function that handles changes to our Secret
+func secretObserver(ctx context.Context, secret *corev1.Secret) {
+	logger := logging.FromContext(ctx)
+
+	if secret == nil {
+		logger.Warn("Nil Secret passed to secretObserver; ignoring")
+		return
+	}
+
+	if dispatcher == nil {
+		// This typically happens during startup
+		logger.Debug("Dispatcher is nil during call to secretObserver; ignoring changes")
+		return
+	}
+
+	// Toss the new secret to the dispatcher for inspection and action
+	newDispatcher := dispatcher.SecretChanged(ctx, secret)
+	if newDispatcher != nil {
+		// The configuration change caused a new dispatcher to be created, so switch to that one
+		logger.Info("Producer Reconfigured; Switching To New Producer")
+		dispatcher = newDispatcher
```

Comment:
Copy/paste issue?  Should "Producer" be "Dispatcher" ?  Also is this the right terminology?  Typically Receiver/Dispatcher go together OR Producer/ConsumerGroup depending on whether it's k8s components or kafka resources - make sure this and receiver version are correct please ; )

Reply:
Changed to Dispatcher.  Also modified the internal messages to refer to closing and restarting Producer/Consumer Groups (the ones from the observer are still Receiver/Dispatcher).  I'm not completely certain those are the best places to use those terms but I attempted some more consistency.